src/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// Represents a set of <see cref="Pipe"/> options
+    /// </summary>
+    public class PipeOptions
+    {
+        /// <summary>
+        /// Default instance of <see cref="PipeOptions"/>
+        /// </summary>
+        public static PipeOptions Default { get; } = new PipeOptions();
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PipeOptions"/>
+        /// </summary>
+        public PipeOptions(
+            MemoryPool<byte> pool = null,
+            PipeScheduler readerScheduler = null,
+            PipeScheduler writerScheduler = null,
+            long pauseWriterThreshold = 0,
+            long resumeWriterThreshold = 0,
+            int minimumSegmentSize = 2048)
+        {
+
+            if (resumeWriterThreshold < 0 && resumeWriterThreshold > pauseWriterThreshold)
```

Comment:
`if ((ulong)resumeWriterThreshhold > pauseWriterThreshold)` ?

Reply:
Couldn't jit do this optimization instead of making code less readable?

Reply:
cc @AndyAyersMS

Reply:
The jit can't do this for you today. It requires recognizing a specific pattern of facts within a specific pattern of compares.

Unless you think this code is super perf critical I would leave it as is. Presumably all these exceptional cases are rare, the computations feeding them are trivial, and so HW branch prediction should handle them pretty cheaply.

Reply:
Added dotnet/coreclr#16363 tracking possible jit improvement.